pre_commit/repository.py

```diff
@@ -64,11 +64,10 @@ def languages(self):
 
     @cached_property
     def additional_dependencies(self):
-        dep_dict = defaultdict(lambda: defaultdict(set))
+        dep_dict = defaultdict(lambda: defaultdict(list))
         for _, hook in self.hooks:
-            dep_dict[hook['language']][hook['language_version']].update(
-                hook.get('additional_dependencies', []),
-            )
+            dep_dict[hook['language']][hook['language_version']] += \
+                hook.get('additional_dependencies', [])
```

Comment:
So we do lose set semantics here, can you add a test with multiple hooks that have the same additional dependencies and verify that there are not duplicates?